projects/core/src/lib/map/services/application-layer-refresh.service.ts

```diff
@@ -0,0 +1,61 @@
+import { DestroyRef, Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { selectAutoRefreshableLayers } from '../state/map.selectors';
+import { Subscription, timer } from 'rxjs';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { LayerManagerModel, MapService } from '@tailormap-viewer/map';
+import { ExtendedAppLayerModel } from '../models';
+
+interface RefreshableLayer extends ExtendedAppLayerModel {
+  id: string;
+  autoRefreshInSeconds: number;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApplicationLayerRefreshService {
+
+  private refreshingLayers = new Map<string, Subscription>();
+  private layerManager: LayerManagerModel | undefined;
+
+  constructor(
+    private store$: Store,
+    private destroyRef: DestroyRef,
+    private mapService: MapService,
+  ) {
+    this.mapService.getLayerManager$()
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe(layerManager => this.layerManager = layerManager);
+    this.store$.select(selectAutoRefreshableLayers)
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe(layers => {
+        const refreshableLayers: RefreshableLayer[] = layers
+          .filter((l): l is RefreshableLayer => typeof l.autoRefreshInSeconds === 'number');
+        this.setRefreshLayers(refreshableLayers);
+      });
+  }
+
+  public setRefreshLayers(refreshableLayer: RefreshableLayer[]) {
+    const ids = new Set(refreshableLayer.map(r => r.id));
+    const currentIds = Array.from(this.refreshingLayers.keys());
+    currentIds.forEach((key) => {
+      if (!ids.has(key)) {
```

Comment:
Once turned on, the refresh time can't be changed (like opacity can) unless layer is hidden again, but we can always add that in the future because we don't have a control like opacity at the moment anyway.